test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -1840,58 +1846,34 @@ def test_init_from_local_shards_and_global_metadata(self):
-            rank1_shard_metadata = ShardMetadata(
-                shard_offsets=[0, 0],
-                shard_sizes=[5, 5],
-                placement="rank:0/cuda:1"
-            )
-            rank3_shard_metadata = ShardMetadata(
-                shard_offsets=[5, 0],
+        if self.rank != 0:
+            local_shard_metadata = ShardMetadata(
+                shard_offsets=[5 * (self.rank - 1), 0],
                 shard_sizes=[5, 5],
-                placement="rank:2/cuda:3"
+                placement=f"rank:{self.rank - 1}/cuda:{self.rank}"
             )
-
-
-            local_shard_metadata = rank1_shard_metadata if self.rank == 1 else rank3_shard_metadata
             local_shards = [_sharded_tensor.Shard(torch.randn(5, 5, device=f"cuda:{self.rank}"), local_shard_metadata)]
 
-            sharded_tensor = _sharded_tensor.init_from_local_shards(local_shards, [10, 5], new_pg, init_rrefs=True)
+            sharded_tensor = _sharded_tensor.init_from_local_shards(local_shards, [15, 5], process_group=new_pg)
 
             # Verify local shard.
             local_shard = sharded_tensor.local_shards()[0]
             self.assertEqual(torch.device(f'cuda:{self.rank}'), local_shard.tensor.device)
             self.assertEqual((5, 5), local_shard.tensor.size())
 
             # Verify local shard metadata.
-            self.assertEqual((self.rank // 2 * 5, 0), local_shard.metadata.shard_offsets)
+            self.assertEqual(((self.rank - 1) * 5, 0), local_shard.metadata.shard_offsets)
             self.assertEqual((5, 5), local_shard.metadata.shard_sizes)
             self.assertEqual(f'rank:{self.rank - 1}/cuda:{self.rank}', str(local_shard.metadata.placement))
 
             # Verify global metadata.
             sharded_tensor_metadata = sharded_tensor.metadata()
             shards_metadata = sharded_tensor_metadata.shards_metadata
-            self.assertEqual(2, len(shards_metadata))
+            self.assertEqual(3, len(shards_metadata))
             for rank, shard_metadata in enumerate(shards_metadata):
                 self.assertEqual((rank * 5, 0), shard_metadata.shard_offsets)
                 self.assertEqual((5, 5), shard_metadata.shard_sizes)
-                self.assertEqual(f'rank:{rank * 2}/cuda:{rank * 2 + 1}', str(shard_metadata.placement))
-
-            # Validate remote shards.
-            remote_shards = sharded_tensor.remote_shards()
-            if self.rank == 1 or self.rank == 3:
-                self.assertEqual(1, len(remote_shards))
-            else:
-                self.assertEqual(2, len(remote_shards))
-
-            owners = {}
-            for rpc_rank, shards in remote_shards.items():
-                self.assertEqual(1, len(shards))
-
-                for remote_shard in shards:
-                    self.assertEqual(rpc_rank, remote_shard.owner().id)
-                    shard = remote_shard.to_here()
-                    self.assertEqual((5, 5), shard.tensor.size())
```

Comment:
Don't we need to keep this validation for remote_shards?

Reply:
I think it's a bit tricky, the reason that I disabled this remote shards validation for this new_group test is that, we are essentially having two pg (default_pg, new_pg) for the test. When initializing sharded tensors, we only initialize (call the constructor) in ranks of the new_pg, but the `_init_rpc` call in `post_init()` will try to all_gather in the default_pg, in this case, rank 0 does not initialize(call constructor), so it does not have this all_gather, thus it hangs. 

https://github.com/pytorch/pytorch/blob/eb2a803406397e884c0d7999f782fa82e0204348/torch/distributed/_sharded_tensor/api.py#L220

I can think of a solution on this but not entirely sure the correct semantic:

we use `sharded_tensor.process_group.all_gather` instead of `rpc.api._all_gather` here, so that it only broadcast the sharded tensor ids in the new_pg, later we still share `all_tensor_ids` in default_pg world.

Reply:
@pritamdamania87 I thought about this more and the solution i proposed might not work, as for those ranks who don't know about sharded tensor initialization, it does not have the `_register_remote_shards` or sharded tensor map on it at all. In this case, I am thinking that can we only broadcast and register remote shards in the `new_group` instead of the default_group?

https://github.com/pytorch/pytorch/blob/e963b436913f8324a12f8ff0987be7b4085020cb/torch/distributed/_sharded_tensor/api.py#L634

Reply:
@wanchaol Yes I do feel we should do the all_gather only within the ranks of the `new_group`. Would this work out or are there other issues? I think we can potentially fix this in a follow PR to unblock landing this critical fix for the test suite.

Reply:
yeah i think this would work out, i will work on a follow up PR after this land.